build.xml

```diff
@@ -8,119 +8,268 @@
+    
+  </target>
+  
+  
+  <target name="compileSchematron" depends="locateSchematron" if="${test.schematron}">
+    <echo message="START COMPILING SCHEMATRON FILE '${xspec.schematron.absolute}'"/>
+    
+    <echo message="   STEP 1: Including modules"/>
+    
+    <xslt in="${xspec.schematron.absolute}" 
+      out="${xspec.dir}/${xspec.base}_dsdl_included.sch" 
+      style="${xspec.project.dir}/src/schematron/iso-schematron/iso_dsdl_include.xsl" 
+      force="true"
+      >
+      <factory name="net.sf.saxon.TransformerFactoryImpl"/>
+    </xslt>
+    
+    <echo message="   STEP 2: Expanding abstracts"/>
+    <xslt in="${xspec.dir}/${xspec.base}_dsdl_included.sch" 
+      out="${xspec.dir}/${xspec.base}.sch" 
+      style="${xspec.project.dir}/src/schematron/iso-schematron/iso_abstract_expand.xsl" 
+      force="true"
+      >
+      <factory name="net.sf.saxon.TransformerFactoryImpl"/>
+    </xslt>
+    
+    <echo message="   STEP 3: Convert Schematron to XSL"/>    
+    <xslt 
+      in="${xspec.dir}/${xspec.base}.sch"
+      out="${xspec.compiled.xsl}" 
+      style="${xspec.project.dir}/src/schematron/iso-schematron/iso_svrl_for_xslt2.xsl" 
+      force="true"
+      >
+      <factory name="net.sf.saxon.TransformerFactoryImpl"/>
+      <param name="allow-foreign" expression="true"/>
+      <param name="phase" expression="${xspec.phase}"/>
+    </xslt>
+    
+    <echo message="   STEP 4: Convert XSpec (into a format that references the XSL generated from the Schematron)"/>    
+    <xslt 
+      in="${xspec.xml}"
+      out="${xspec.compiled.xml}" 
+      style="${xspec.project.dir}/src/schematron/schut-to-xspec.xsl" 
+      force="true"
+      >
+      <factory name="net.sf.saxon.TransformerFactoryImpl"/>
+      <param name="stylesheet" expression="${xspec.compiled.xsl.param}"/>
+    </xslt>
+
+    <echo message="COMPILATION COMPLETE"/>
+    
+   </target>
+  
+   
 
+  
   <target name="xspec"
-    description="Runs the XSpec unit tests in ${xspec.xml}">
-    <xslt in="${xspec.xml}" 
+    description="Runs the XSpec unit tests in ${xspec.compiled.xml}" depends="copyXSpec,compileSchematron">
+    
+    <echo message="Creating Test Stylesheet..."/>
+    <xslt in="${xspec.compiled.xml}" 
       out="${xspec.xsl}"
       style="${xspec.project.dir}/src/compiler/generate-xspec-tests.xsl"
-      classpath="${saxon.jar}">
-			<factory name="net.sf.saxon.TransformerFactoryImpl" />
-		</xslt>
-		<!-- Can't specify initial template with <xslt> task so have to
-		     fall back to running Saxon as Java app. -->
-		<java classname="net.sf.saxon.Transform"
-					classpath="${saxon.jar}"
-					fork="true">
-			<arg value="-ext:on" />
-			<arg value="-s:${xspec.xml}" />
-			<arg value="-o:${xspec.result.xml}" />
-			<arg value="-xsl:${xspec.xsl}" />
-			<arg value="-it:{http://www.jenitennison.com/xslt/xspec}main" />
-		</java>
+      force="true"
+      >     
+      <factory name="net.sf.saxon.TransformerFactoryImpl" />     
+    </xslt>
```

Comment:
oXygen's `build.xml` enables XML catalog here and in other places. Whey are they not incorporated in this PR?

Reply:
I reintroduced XML catalog in various places in https://github.com/xspec/xspec/commit/3bcb2cd3ecaa6ff117e3193546673941544eff4f following the current version of `build.xml` packaged with oXygen 19. I had to add `xml-resolver.jar` as additional jar library in `.travis.yaml` and `appveyor.yaml` otherwise tests failed on my local machine with the following error message:

 ```
[java] Transformation failed: Failed to load Apache catalog resolver library [java]
```

Tests are passing on [Travis](https://travis-ci.org/xspec/xspec/builds/333452222) but failing on [AppVeyor](https://ci.appveyor.com/project/xspec/xspec/build/1.0.334/job/db55wg40d1rhayof). The culprit is the ant test for Schematron which fails with this error:

```
[java] Error on line 12 column 87 of demo-03.xsl:
     [java]   Malformed URL C:/projects/xspec/tutorial/schematron/demo-03-sch-compiled.xsl(base
     [java]   file:/C:/projects/xspec/tutorial/schematron/xspec/demo-03.xsl): unknown protocol: c
     [java] Malformed URL C:/projects/xspec/tutorial/schematron/demo-03-sch-compiled.xsl(base file:/C:/projects/xspec/tutorial/schematron/xspec/demo-03.xsl)
``` 

The issue is the path `file:/C`, I suspect the XML resolver or the java command may have caused this. I also wonder if this is a similar problem I described in https://github.com/xspec/xspec/pull/138#pullrequestreview-53085196 (see where I talk about `file:/` in Unix and Windows) but this time it's problematic for Windows. 

I won't have more time today to debug this and unfortunately Windows is not my forte. Could you have a look at this failing test in case you spot any glaring error I made?

Reply:
Thanks for the improvements!
I looked at the intermediate file (`demo-03.xsl`) and found a fake URI:
```xml
   <xsl:import href="c:/xspec/tutorial/schematron/demo-03-sch-compiled.xsl"/>
...
   <xsl:variable name="x:stylesheet-uri"
                 as="xs:string"
                 select="'c:/xspec/tutorial/schematron/demo-03-sch-compiled.xsl'"/>
```
Probably https://github.com/xspec/xspec/pull/160/commits/4e2ed51f98b7f530b21bbbc3742eb97fe9712294#diff-2cccd7bf48b7a9cc113ff564acd802a8R129 makes it.
I'm fixing it.